src/main/java/dev/aziz/movies/services/MovieService.java

```diff
@@ -39,9 +47,41 @@ public MovieDto createMovie(MovieDto movieDto) {
         if (movieRepository.existsMovieByTitle(movieDto.getTitle())) {
             throw new AppException("Movie already exists.", HttpStatus.BAD_REQUEST);
         }
-        return movieMapper.movieToMovieDto(movieRepository.save(movieMapper.movieDtoToMovie(movieDto)));
+
+        List<Genre> genres = movieDto.getGenres().stream()
+                .map(genreDto -> genreRepository.findGenreByName(genreDto.getName())
+                        .orElseGet(() -> genreRepository.save(
+                                        Genre.builder().name(genreDto.getName()).build()
```

Comment:
You should use mappers to go from a `GenreDto` to `Genre`. Same for `ActorDto` to `Actor`, and `DirectorDto` to `Director`.

For both `Actor` and `Director`, create a separated mapper instead of using the `MovieMapper`.